arc/src/main.c

```diff
@@ -762,55 +768,132 @@ static int stop_gyro_trigger(struct device *dev)
     return 0;
 }
 
+#ifdef BUILD_MODULE_SENSOR_LIGHT
+static double cube_root_recursive(double num, double low, double high) {
```

Comment:
Personally I think we should be returning only raw data from sensors and doing conversions in the JavaScript.  Otherwise I fear we are going to start adding too much code specific to certain sensors to ZJS.  Now that we have the ability to use JavaScript modules I think keeping the sensor specific code out of the native space makes sense.  Thoughts?